src/auto/tree_row_reference.rs

```diff
@@ -0,0 +1,71 @@
+// This file was generated by gir (4d68d19) from gir-files (11e0e6d)
+// DO NOT EDIT
+
+use TreeModel;
+use TreePath;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+
+glib_wrapper! {
+    pub struct TreeRowReference(Boxed<ffi::GtkTreeRowReference>);
+
+    match fn {
+        copy => |ptr| ffi::gtk_tree_row_reference_copy(mut_override(ptr)),
+        free => |ptr| ffi::gtk_tree_row_reference_free(ptr),
+    }
+}
+
+impl TreeRowReference {
+
+    pub fn new<T: IsA<TreeModel>>(model: &T, path: &TreePath) -> TreeRowReference {
```

Comment:
I was hoping it wasn't the case but we'll have to make the generator respect `nullable` in constructors (https://github.com/gtk-rs/gir/issues/289). The assertions are there behind the scenes already but it'd be more consistent with other treeview-path-related functions to return `Option`. We'll have to fix `TreePath::new_from_string` and other similar cases as well.

The vast majority of constructors are infallible though, that's why the generator doesn't handle this case.